pkg/applications/applications.go

```diff
@@ -0,0 +1,27 @@
+package applications
+
+import (
+	v1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetApplications returns a list of SourceRepositories
+func GetApplications(jxClient versioned.Interface, ns string, envMap map[string]*v1.Environment) ([]v1.SourceRepository, error) {
```

Comment:
Thanks @hferentschik this is very useful feedback! Yes I agree, it makes more sense to do it the way you describe. I was a bit hesitant to move all that logic into applications pkg because I felt we were probably still going to use some of it in the cobra command but it also felt wrong to have to give it all these params. I guess we should make this first step and then refactor whatever is missing in the cobra command in the future.